core/src/main/kotlin/pages/contentNodeProperties.kt

```diff
@@ -7,10 +7,40 @@ class SimpleAttr(val extraKey: String, val extraValue: String) : ExtraProperty<C
     override val key: ExtraProperty.Key<ContentNode, SimpleAttr> = SimpleAttrKey(extraKey)
 
     companion object {
-        fun header(value: String) = SimpleAttr("data-togglable", value)
+        fun togglableTarget(value: String) = SimpleAttr(TOGGLEABLE_CONTENT_TYPE_ATTR, value)
     }
 }
 
+object HtmlInvisibleExtra : ExtraProperty<ContentNode>, ExtraProperty.Key<ContentNode, HtmlInvisibleExtra> {
+    override val key: ExtraProperty.Key<ContentNode, *> = this
+}
+
+const val TOGGLEABLE_CONTENT_TYPE_ATTR = "data-togglable"
+enum class BasicToggleableContentType : ToggleableContentType {
+    TYPE, CONSTRUCTOR, FUNCTION, PROPERTY, ENTRY, EXTENSION, INHERITED_FUNCTION, INHERITED_PROPERTY
+}
+
+interface ToggleableContentType
+class ToggleableContentTypeExtra(val value: ToggleableContentType) : ExtraProperty<ContentNode> {
+    companion object : ExtraProperty.Key<ContentNode, ToggleableContentTypeExtra>
+    override val key: ExtraProperty.Key<ContentNode, ToggleableContentTypeExtra> = ToggleableContentTypeExtra
+}
+
+/**
+ * @param text a tab name
+ */
+data class ContentTab(val text: ContentText, val toggleableContentTypes: List<ToggleableContentType>)
```

Comment:
> Is ContentText that requires DCI and source sets really needed, or it can be just a simple String?

Not really, but it left a possibility to custom a tab text.

> If it's really needed, what happens if the user uses the wrong DCI content kind or the wrong source sets?

Wrong DCI or source sets can be everywhere.  Although, it is not easy to define what is the "right" DCI content kind for the tab. Should it be the same as tabbed group or only a tab content?

Reply:
> Not really, but it left a possibility to custom a tab text.

What do you mean by this? Customize in what way?

> Although, it is not easy to define what is the "right" DCI content kind for the tab. Should it be the same as tabbed group or only a tab content?

Yes, it's unclear, that is exactly why I'm asking what happens if the API consumer chooses some random DCI and if it can be replaced by a simple `String` :)

Reply:
> What do you mean by this? Customize in what way?

at least, styles, multiline text. for example, to add an icon.

> Yes, it's unclear, that is exactly why I'm asking what happens if the API consumer chooses some random DCI and if it can be replaced by a simple String :)

It does not matter. Or what do you mean?

Reply:
> It does not matter.

Oh. Yeah, that's not clear that it doesn't matter which values you pass in there. Generally speaking, having parameters that do nothing is not good in my opinion, especially if they are as complicated as DCI/sourceSet.

I actually expected that if I passed `Text(.., sourceSet = X)`, the header would be shown only if I chose `X` in the source set tab.